src/Core/src/Hosting/AppHostBuilder.cs

```diff
@@ -76,7 +64,11 @@ public IAppHost Build()
 
 			ConfigureServiceCollectionBuilders(_serviceProvider);
 
-			return new AppHost(_serviceProvider, null);
+			var appHost = new Internal.AppHost(_serviceProvider, null);
+
+			AppHost.Current = appHost;
```

Comment:
I am setting current here, so this should not require modification for additional platforms.

Reply:
Should we add a check here to make sure we can't set Current again ?